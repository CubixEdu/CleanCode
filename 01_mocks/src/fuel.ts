export class Fuel {

    public fill(litersOfFuel: number) {
        return litersOfFuel;
    }
}