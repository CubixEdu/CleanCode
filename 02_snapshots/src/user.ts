export class User {
  
    constructor(
      private id: number,
      private username: string,
      private email: string,
      private firstName: string,
      private lastName: string,
      private dateOfBirth: Date,
      private address: {
        street: string;
        city: string;
        zipCode: string;
      },
      private phoneNumber: string,
      private isActive: boolean,
      private roles: string[]
    ) {}
  }