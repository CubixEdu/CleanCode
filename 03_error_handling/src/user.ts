export class User {
  
    constructor(
      private id: number,
      private username: string,
      private email: string,
      private firstName: string,
      private lastName: string,
      private dateOfBirth: Date
    ) {}
  }