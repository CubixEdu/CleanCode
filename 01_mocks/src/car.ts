import { Fuel } from "./fuel";

export class Car {

    constructor(private fuel: Fuel) { }

    public drive() {
        this.fuel.fill(35);
    }
}