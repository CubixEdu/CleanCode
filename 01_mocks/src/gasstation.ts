import { Fuel } from "./fuel";

export class GasStation {

    constructor(private fuel: Fuel) { }

    public fillUpCars(numberOfCars: number) {
        for (let index = 1; index < numberOfCars + 1; index++) {
            this.fuel.fill(index + 1);
        }
    }
}