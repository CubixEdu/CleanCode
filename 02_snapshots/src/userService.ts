import { User } from "./user";

export class UserService {

  public getUserById(id: number) {
    // simulate a real user data retrieval here
    return new User(
      id,
      'john_doe',
      'john@example.com',
      'John',
      'Doe',
      new Date('1990-01-15'),
      {
        street: '123 Main St',
        city: 'Anytown',
        zipCode: '12345',
      },
      '123-456-7890',
      true,
      ['user', 'admin']
    );
  }
}
