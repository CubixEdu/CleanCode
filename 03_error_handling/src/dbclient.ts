import { User } from "./user";

export class DbClient {

    public getUserByUsername(username: string) {
        return new User(
            1,
            'john_doe',
            'john@example.com',
            'John',
            'Doe',
            new Date('1990-01-15')
          );
    }
}