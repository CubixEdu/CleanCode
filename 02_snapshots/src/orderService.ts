import { UserService } from "./userService";

export class OrderService {

    constructor(private userService: UserService) { }

    public getUsersOfTheLastHour() {
        const users = [];
        for (let index = 1; index < 5; index++) {
            const user = this.userService.getUserById(index);
            users.push(user);
        }
        return users;
    }
}