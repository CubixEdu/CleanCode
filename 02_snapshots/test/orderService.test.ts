import { OrderService } from "../src/orderService"
import { User } from "../src/user";
import { UserService } from "../src/userService";
import { mock, mockReset } from 'jest-mock-extended';

const mockedUserService = mock<UserService>();

describe('OrderService tests', () => {

    it('should return all the users who made an order in the last hour', () => {
        // Arrange
        const dummyUser = new User(
            1,
            'john_doe',
            'john@example.com',
            'John',
            'Doe',
            new Date('1990-01-15'),
            {
                street: '123 Main St',
                city: 'Anytown',
                zipCode: '12345',
            },
            '123-456-7890',
            true,
            ['user', 'admin']
        );
        mockedUserService.getUserById.mockReturnValue(dummyUser);
        const sut = new OrderService(mockedUserService);

        // Act
        const actualResult = sut.getUsersOfTheLastHour();

        // Assert
        expect(actualResult).toMatchSnapshot();
        expect(mockedUserService.getUserById).toMatchSnapshot();
    })
})