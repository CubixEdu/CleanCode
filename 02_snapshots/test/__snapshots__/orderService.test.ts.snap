// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderService tests should return all the users who made an order in the last hour 1`] = `
[
  User {
    "address": {
      "city": "Anytown",
      "street": "123 Main St",
      "zipCode": "12345",
    },
    "dateOfBirth": 1990-01-15T00:00:00.000Z,
    "email": "john@example.com",
    "firstName": "John",
    "id": 1,
    "isActive": true,
    "lastName": "Doe",
    "phoneNumber": "123-456-7890",
    "roles": [
      "user",
      "admin",
    ],
    "username": "john_doe",
  },
  User {
    "address": {
      "city": "Anytown",
      "street": "123 Main St",
      "zipCode": "12345",
    },
    "dateOfBirth": 1990-01-15T00:00:00.000Z,
    "email": "john@example.com",
    "firstName": "John",
    "id": 1,
    "isActive": true,
    "lastName": "Doe",
    "phoneNumber": "123-456-7890",
    "roles": [
      "user",
      "admin",
    ],
    "username": "john_doe",
  },
  User {
    "address": {
      "city": "Anytown",
      "street": "123 Main St",
      "zipCode": "12345",
    },
    "dateOfBirth": 1990-01-15T00:00:00.000Z,
    "email": "john@example.com",
    "firstName": "John",
    "id": 1,
    "isActive": true,
    "lastName": "Doe",
    "phoneNumber": "123-456-7890",
    "roles": [
      "user",
      "admin",
    ],
    "username": "john_doe",
  },
  User {
    "address": {
      "city": "Anytown",
      "street": "123 Main St",
      "zipCode": "12345",
    },
    "dateOfBirth": 1990-01-15T00:00:00.000Z,
    "email": "john@example.com",
    "firstName": "John",
    "id": 1,
    "isActive": true,
    "lastName": "Doe",
    "phoneNumber": "123-456-7890",
    "roles": [
      "user",
      "admin",
    ],
    "username": "john_doe",
  },
]
`;

exports[`OrderService tests should return all the users who made an order in the last hour 2`] = `
[MockFunction] {
  "calls": [
    [
      1,
    ],
    [
      2,
    ],
    [
      3,
    ],
    [
      4,
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": User {
        "address": {
          "city": "Anytown",
          "street": "123 Main St",
          "zipCode": "12345",
        },
        "dateOfBirth": 1990-01-15T00:00:00.000Z,
        "email": "john@example.com",
        "firstName": "John",
        "id": 1,
        "isActive": true,
        "lastName": "Doe",
        "phoneNumber": "123-456-7890",
        "roles": [
          "user",
          "admin",
        ],
        "username": "john_doe",
      },
    },
    {
      "type": "return",
      "value": User {
        "address": {
          "city": "Anytown",
          "street": "123 Main St",
          "zipCode": "12345",
        },
        "dateOfBirth": 1990-01-15T00:00:00.000Z,
        "email": "john@example.com",
        "firstName": "John",
        "id": 1,
        "isActive": true,
        "lastName": "Doe",
        "phoneNumber": "123-456-7890",
        "roles": [
          "user",
          "admin",
        ],
        "username": "john_doe",
      },
    },
    {
      "type": "return",
      "value": User {
        "address": {
          "city": "Anytown",
          "street": "123 Main St",
          "zipCode": "12345",
        },
        "dateOfBirth": 1990-01-15T00:00:00.000Z,
        "email": "john@example.com",
        "firstName": "John",
        "id": 1,
        "isActive": true,
        "lastName": "Doe",
        "phoneNumber": "123-456-7890",
        "roles": [
          "user",
          "admin",
        ],
        "username": "john_doe",
      },
    },
    {
      "type": "return",
      "value": User {
        "address": {
          "city": "Anytown",
          "street": "123 Main St",
          "zipCode": "12345",
        },
        "dateOfBirth": 1990-01-15T00:00:00.000Z,
        "email": "john@example.com",
        "firstName": "John",
        "id": 1,
        "isActive": true,
        "lastName": "Doe",
        "phoneNumber": "123-456-7890",
        "roles": [
          "user",
          "admin",
        ],
        "username": "john_doe",
      },
    },
  ],
}
`;
